Lib/test/support/__init__.py

```diff
@@ -1999,3 +1999,12 @@ def check_disallow_instantiation(testcase, tp, *args, **kwds):
         qualname = f"{name}"
     msg = f"cannot create '{re.escape(qualname)}' instances"
     testcase.assertRaisesRegex(TypeError, msg, tp, *args, **kwds)
+
+@contextlib.contextmanager
+def infinite_recursion(max_depth=75):
```

Comment:
Minor suggestion: I think adding a docstring would help others know where/why they need this.

Reply:
Ah, missed this one! Yeap, forgot to do that. Will add a follow up PR.